chainer/optimizers/adam.py

```diff
@@ -102,8 +119,15 @@ def __init__(self, parent_hyperparam=None,
             self.hyperparam.weight_decay_rate = weight_decay_rate
         if amsgrad is not None:
             self.hyperparam.amsgrad = amsgrad
+        if adabound is not None:
+            self.hyperparam.adabound = adabound
+        if final_lr is not None:
+            self.hyperparam.final_lr = final_lr
+        if gamma is not None:
+            self.hyperparam.gamma = gamma
 
     def init_state(self, param):
+        self.state['initial_alpha'] = self.hyperparam.alpha
```

Comment:
As of now, since only AdaBound uses `self.state['initial_alpha']`, so how about set this only when self.hyperparam.alpha is True?

e.g.
```python
if self.hyperparam.adabound:
    self.state['initial_alpha'] = self.hyperparam.alpha
```